### PR TITLE
Feature/google api calender

### LINKED ISSUE #39: Google calendar backend
## Goal
- To manage sauna availability, bookings should be added to a Google Calendar
- Users will then see, before booking, if the sauna is still available at the intended time-slot
- It would probably be the easiest to have one Google Calendar per sauna

## Tasks
- [x] Provide a class / package that allows to easily insert a booking into a Google Calendar, presumably using [Google Calendar API for Java](https://developers.google.com/calendar/api/quickstart/java)
- [x] Return some sort of ID when the event is inserted into the event, such that a reference to the calendar entry can be stored into the database and found in the Google Calendar again later
- [x] Provide some method to update an event in the Google Calendar by passing the ID of the event
- [x] Provide some method to delete an event in the Google Calendar by passing the ID of the event
- [x] Make the class generic, such that it works for different calendars → as probably every sauna will have a different calendar, this should work for multiple calendars, probably by passing something like the ID of the calendar

### PR DESCRIPTION
**Pull request by Utschu**
## Summary
Create an interface to communicate with the google API  Service
## Related Issues

- Closes #39
- Fixes #

## Definition of Done

- User story
  - [x] All stated conditions fulfilled
- Code
  - [x] No more code needed
  - [x] No known bugs
- Clean Code
  - [x] JavaDoc has been added
  - [x] API documentation has been added via annotations (if applicable)
  - [x] No unavoidable code smells
- Testing
  - [x] All existing tests pass
  - [x] New unit tests created according to [CONTRIBUTING.md](./CONTRIBUTING.md#-testing-strategy)
  - [x] New unit tests pass
- SonarCloud
  - [x] Quality gate passed
- Compatibility to frontend
  - [ ] Compatibility to staging-frontend verified (running frontend locally in Docker container with image matching the deployed image on staging)

### CODE REVIEW

File: src/test/java/ch/saunah/saunahbackend/service/GoogleCalendarServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    /**
+     * Test if the authenicate is functioning
+     * @throws IOException
+     */
+    @Test
+    void testDemoListEvents() throws IOException {
+        List<Event> events = calendarService.getNextEvents(TEST_CALENDAR_ID);
+
+        assertNotNull(events);
+        System.out.println(events);
```
Comment (TG134516) [inline src/test/java/ch/saunah/saunahbackend/service/GoogleCalendarServiceTest.java:40]: Remove system.out.println if they are no longer necessary

**Review by TG134516 — DISMISSED** [review]
small change required and solve merge conflict.

**Review by simonschuhmacher — CHANGES_REQUESTED** [review]
Looks good to me 🙂
I just cleaned up a bit for the gradle dependencies (not directly related to this pr) and removed stuff we no longer needed and updated some dependencies to the latest version.

Except of the print statement marcel mentioned, I think it's nice.

**Review by simonschuhmacher — APPROVED** [review]
Nice

**Pull request merged**